jscomp/bsb/bsb_ninja_rule.ml

```diff
@@ -161,9 +164,18 @@ let make_custom_rules ~(gentype_config : Bsb_config_types.gentype_config option)
     | None -> ()
     | Some flag ->
         Ext_buffer.add_char_string buf ' ' (Bsb_build_util.pp_flag flag));
-    (match (has_reason_react_jsx, reason_react_jsx) with
-    | false, _ | _, None -> ()
-    | _, Some Jsx_v3 -> Ext_buffer.add_string buf " -bs-jsx 3");
+    (match (has_reason_react_jsx, reason_react_jsx, jsx_version) with
+    | _, _, Some Jsx_v3 -> Ext_buffer.add_string buf " -bs-jsx 3"
+    | _, _, Some Jsx_v4 -> Ext_buffer.add_string buf " -bs-jsx 4"
+    | _, Some Jsx_v3, None -> Ext_buffer.add_string buf " -bs-jsx 3"
```

Comment:
Not sure I understand this case.
Why not handle the 2  cases separately `reason_react_jsx` and `jsx_version` and add the options. (And if things are repeated the option is repeated). Seems more predictable than these specific cases used here.

Reply:
I'm afraid the options are repeated. And two options`reason_react_jsx` and `jsx_version` exist temporarily, I think it is more proper to handle at the same match. Because these two options are making the one option anyway.

Reply:
I think having a default of jsx 3 (inside bsc) and ignoring `reason_react_jsx` has the same effect as what the pattern match is trying to do right now.

Reply:
Not sure I understand your intention correctly. I'm afraid we can't identify the pattern that `reason_react_jsx` is None, if we use jsx 3 as a default value.
```rescript
match jsx_version with
| Some Jsx_v3 ->
| Some Jsx_v4 ->
| None ->
// we are missing the pattern both options are none -> not activate the jsx ppx
```

Reply:
What I'm saying is that `reason_react_jsx` is ignored already now. So just keep on ignoring it.

Reply:
Maybe I missed something. I think if we ignore `reason_react_jsx` value and have jsx 3 as a default value here, V3 will be triggered no matter whether the configuration has "reason.react-jsx" or not in the existing projects.

Reply:
That's correct the option is useless as you can only select 1 version.

So one can switch to v4 by using the new JSX config.
Old config can be ignored as there's nothing useful to specify with it.

Reply:
I got your point, your intention is to make users to use the new JSX config with V10, right? If so, ignoring `reason_react_jsx` option makes sense to me. One thing in mind is if the user uses V10 on the project with dependency of old config(no `reason_react_jsx` config at all in there). In this case the dependency will be built with V3 ppx activated. It that alright?

Reply:
I think the config of the top-level project determines the config of all the sub-projects. But one would have to check to be sure. There's some doc on pinned dependencies that might help indirectly answer that question.

Reply:
related https://github.com/rescript-lang/rescript-compiler/issues/5515